app/questionnaire/questionnaire_schema.py

```diff
@@ -335,20 +336,14 @@ def _populate_answer_dependencies(self) -> None:
                     self._update_answer_dependencies_for_calculations(
                         question["calculations"], block_id=block["id"]
                     )
+                    self.update_dependencies_for_dynamic_answers(
```

Comment:
just occured to me, if this needs to run when question is calculated and when it isn't could you put this update before the if block and then you don't need it twice?

Reply:
Makes absolute sense, refactored now, thanks!

Reply:
at a glance I'm not sure what has gone wrong with the test when this was updated, I didn't think the answer dependencies for calculations would depend on going first? if it did.... oops

Reply:
functionally it still works fine 🤔

Reply:
test passes locally too